include/klee/ADT/ImmutableMap.h

```diff
@@ -44,6 +44,26 @@ template <class K, class D, class CMP = std::less<K>> class ImmutableMap {
     return *this;
   }
 
+  bool operator==(const ImmutableMap &b) const {
+    for (iterator it1 = begin(), it2 = b.begin();
+         it1 != end() && it2 != b.end(); ++it1, ++it2) {
+      if (*it1 != *it2)
+        return false;
+    }
+    return size() == b.size();
```

Comment:
Swap the size equality condition with the element equality conditions for better performance